### PR TITLE
breaking lines in boutiques param descriptors resolves #1150

### LINKED ISSUE #1150: Boutiques parameter descriptors with linebreaks
Presently the line breaks in boutiques parameter descriptors are ignored, but users ask for line break support. This will help to better structure long parameters.

### PR DESCRIPTION
**Pull request by MontrealSergiy**
Darius would like to better format long parameter descriptors in PhysIO, so asked to add support for line breaks

https://github.com/dariusvalevicius/PhysIO_CBRAIN/blob/master/cbrain_task_descriptors/tapasphysio.json

### CODE REVIEW

**Comment by prioux** [comment]
Why are you including erb and ERB::Util at all these places?!?? These were never necessary before.

**Comment by MontrealSergiy** [comment]
To avoid error messages for undefined ` h```  method , namely when running rake

Should I put ERB::Util somewhere else or what?

> Why are you including erb and ERB::Util at all these places?!?? These were never necessary before.

File: BrainPortal/lib/cbrain_task_generators/templates/task_params.html.erb.erb [hunk 1]
```diff
@@ -301,7 +301,7 @@
 <%%# Generate a parameter's description block for +desc+ %>
 <%% description = lambda do |desc, isGroupMember = false| %>
   <span class="tsk-prm-desc">
-    <%%= desc %><%%= "<br><br>".html_safe if isGroupMember %>
+    <%%= desc.split("\n").map { |s| h(s) }.join("<br>").html_safe if desc %><<%%= "<br><br>".html_safe if isGroupMember %>
```
Comment (prioux) [inline BrainPortal/lib/cbrain_task_generators/templates/task_params.html.erb.erb:304]: You have a syntax error here, you have one too many `<` (search for `<<`)

File: BrainPortal/lib/cbrain_task_generators/schema_task_generator.rb [hunk 2]
```diff
@@ -21,6 +21,8 @@
 #
 
 require 'fileutils'
+require 'erb'
```
Comment (prioux) [inline BrainPortal/lib/cbrain_task_generators/schema_task_generator.rb:24]: All modifications to `schema_task_generator.rb` are unnecessary

File: BrainPortal/lib/cbrain_task_generators/templates/task_params.html.erb.erb [hunk 3]
```diff
@@ -450,7 +450,9 @@
         <%= name %> <%% groupCheckbox.('<%= id %>',<%= defaultChecked %>) %>
       </h3>
       <h5 style="font-size : 9.5pt; font-weight: normal; padding-top: 2px;" class="grp-desc">
-        <%= groupDescription %>
+        <% groupDescription.split('\n') do |paragraph|%>
```
Comment (prioux) [inline BrainPortal/lib/cbrain_task_generators/templates/task_params.html.erb.erb:453]: That code change does not split lines in the rendered page, there are no new `<br>` introduced

**Review by prioux — CHANGES_REQUESTED** [review]
Please fix all the problems

**Comment by prioux** [comment]
I tried with a plain CBRAIN that introduced ONLY the change at line 304 of `BrainPortal/lib/cbrain_task_generators/templates/task_params.html.erb.erb` for a field that contains a single "\n" in the description. it works if we fix the syntax error:

Before:
Screen Shot 2021-09-13 at 15 27 53
After:
Screen Shot 2021-09-13 at 15 28 47

There was absolutely no need for any of the changes in the schema task generator.

**Comment by MontrealSergiy** [comment]
ok, will do

**Comment by MontrealSergiy** [comment]
Yes, with one line change, no import is needed. I guess it is better than nothing

**Pull request closed without merging**